modules/misc.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+# Copyright 2022 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+function merge_mender_configs() {
+    declare -a ol_targets=()
+    local log="work/mergefiles/log"
+    local tmp="work/mergefiles/tmp"
+
+    for overlay in "${overlays[@]}"; do
+        [ -f "${overlay}/etc/mender/mender.conf" ] && ol_targets+=("${overlay}")
+    done
+
+    mkdir -p work/mergefiles
+    local -r menderconfs="input/resources/mender.conf ${ol_targets[@]/%//etc/mender/mender.conf}"
```

Comment:
Note that `input/resources/mender.conf` might not exist if the user is not running any of the bootstrap scripts.

Reply:
Yes, but we already check it [here](https://github.com/mendersoftware/mender-convert/blob/2221ad3b920e2f9b83be112e271fb100d4aee810/mender-convert-modify#L193-L199).

Well, then the question becomes: does the ticket imply removal of the requirement of running a bootstrap script before mender-convert?

Reply:
But that is just a warning. If you don't run any bootstrap and provide your `mender.conf` through an overlay it should work just fine (while now in this PR it will fail because `input/resources/mender.conf` won't exist).

Reply:
ah omg, i read that wrong. thanks

Reply:
Well, since it was never a requirement to run the bootstrap scripts (which I was under the impression of), lets to what the ticket said: to write mender.conf in /etc/mender and document the fact that at least one mender.conf file is needed from the overlays and that merging overlays are done in the order they are specified (which is the same order of which the entire overlays are applied to rootfs)